src/datetime.ts

```diff
@@ -109,67 +109,43 @@ function tsToObj(ts, offset) {
     minute: d.getUTCMinutes(),
     second: d.getUTCSeconds(),
     millisecond: d.getUTCMilliseconds()
-  };
+  } as GregorianDateTime;
```

Comment:
Instead of using `as` here you could just set the return type on the function signature.

Reply:
That's what I did initially. For consistency with the other methods, I then all the return types in function signatures (the only ones left are [type guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards)).

If that's more idiomatic, I'll switch back to explicit return type.

Reply:
FYI, I reverted these and they are not identical.

Using `return res as X` in a function will indeed define the return type of the function as `X`. But it is brute force, and this `as` is way too strong since one can write `return 42 as string`.

On the other hand, specifying a return type on a method (when it cannot be properly inferred automatically) will also define the method's return type. But this time the compiler will simply check that the value returned indeed matches that type, which is what we usually want.